lib/request_overrider.js

```diff
@@ -1,604 +0,0 @@
-    options.headers[HOST_HEADER] = interceptor.__nock_scopeHost
-    setHeader(req, HOST_HEADER, interceptor.__nock_scopeHost)
-  } else {
-    //  For all other cases, we always add host header equal to the
-    //  requested host unless it was already defined.
-    if (options.host && !getHeader(req, HOST_HEADER)) {
-      let hostHeader = options.host
-
-      if (options.port === 80 || options.port === 443) {
-        hostHeader = hostHeader.split(':')[0]
-      }
-
-      setHeader(req, HOST_HEADER, hostHeader)
-    }
-  }
-}
-
-function RequestOverrider(req, options, interceptors, remove) {
-  const response = new IncomingMessage(new EventEmitter())
-
-  const requestBodyBuffers = []
-  let ended
-  let headers
-
-  //  We may be changing the options object and we don't want those
-  //  changes affecting the user so we use a clone of the object.
-  options = _.clone(options)
-
-  response.req = req
-
-  if (options.headers) {
-    //  We use lower-case header field names throughout Nock.
-    options.headers = common.headersFieldNamesToLowerCase(options.headers)
-
-    headers = options.headers
-    _.forOwn(headers, function(val, key) {
-      setHeader(req, key, val)
-    })
-  }
-
-  /// options.auth
-  if (options.auth && (!options.headers || !options.headers.authorization)) {
-    setHeader(
-      req,
-      'Authorization',
-      `Basic ${Buffer.from(options.auth).toString('base64')}`
-    )
-  }
-
-  req.path = options.path
-  req.method = options.method
-
-  options.getHeader = function(name) {
-    return getHeader(req, name)
-  }
-
-  // ClientRequest.connection is an alias for ClientRequest.socket
-  // https://nodejs.org/api/http.html#http_request_socket
-  // https://github.com/nodejs/node/blob/b0f75818f39ed4e6bd80eb7c4010c1daf5823ef7/lib/_http_client.js#L640-L641
-  // IncomingMessage.connection & IncomingMessage.client are aliases for IncomingMessage.socket
-  // https://nodejs.org/api/http.html#http_response_socket
-  // https://github.com/nodejs/node/blob/b0f75818f39ed4e6bd80eb7c4010c1daf5823ef7/lib/_http_incoming.js#L44-L69
-  // The same Socket is shared between the request and response to mimic native behavior.
-  req.socket = req.connection = new Socket({ proto: options.proto })
-  response.socket = response.client = response.connection = req.socket
-
-  propagate(['timeout'], req.socket, req)
-
-  // Emit a fake socket event on the next tick to mimic what would happen on a real request.
-  // Some clients listen for a 'socket' event to be emitted before calling end(),
-  // which causes nock to hang.
-  process.nextTick(() => {
-    req.emit('socket', req.socket)
-
-    // https://nodejs.org/api/net.html#net_event_connect
-    req.socket.emit('connect')
-
-    // https://nodejs.org/api/tls.html#tls_event_secureconnect
-    if (req.socket.authorized) {
-      req.socket.emit('secureConnect')
-    }
-  })
-
-  req.write = function(buffer, encoding, callback) {
```

Comment:
These implementations were moved to e.g. `handleWrite`, which in `attachToReq()` are set to override e.g. `req.write`.